Source/CSharpEssentials/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessCodeFixProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace CSharpEssentials.InvokeDelegateWithConditionalAccess
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InvokeDelegateWithConditionalAccessCodeFixProvider)), Shared]
+    internal class InvokeDelegateWithConditionalAccessCodeFixProvider : CodeFixProvider
+    {
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticIds.InvokeDelegateWithConditionalAccessId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                "Delegate invocation can be simplified",
+                async c => await UpdateDocumentAsync(context).ConfigureAwait(false),
+               equivalenceKey: nameof(InvokeDelegateWithConditionalAccessCodeFixProvider)),
+               context.Diagnostics);
+            return Task.FromResult(false);
+        }
+
+        private async Task<Document> UpdateDocumentAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var diagnostic = context.Diagnostics.First();
+
+            var localDeclarationLocation = diagnostic.AdditionalLocations[0];
```

Comment:
It might be good to be resilient against the case that the "additional locations" bug is present. It's entirely possible that someone might use C# Essentials without Update 1 RC installed, and I'd hate to see them broken.

*Aside*: I don't *think* there's a way to make the C# Essentials extension show up for Update 1 RC only. I'll look into that though.